workshops/hcload/README.md

```diff
@@ -0,0 +1,465 @@
+
+- Import the "ngrok" module and expose the server to the web (in `mod.ts`)
+
+```js
+import { connect, disconnect } from 'https://deno.land/x/ngrok@2.2.3/mod.ts'
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+
+export default async function (filePath: string): Promise<void> {
+    const fileBase = path.parse(filePath).base
+
+    const app = new Application()
+    app.use(async (context: any) => {
+        await send(context, filePath, { root: '/' })
+    })
+
+    app.addEventListener("listen", async ({ port }) => {
+        const ngrokUrl = `https://${await connect({ protocol: 'http', port })}/${fileBase}`
+        console.log("NGROK: " + ngrokUrl)
+    })
+
+    await app.listen({ port: 20685 })
+}
+```
+
+Here's what fileBase is:
+    - fileBase is the last part of the path to the file, e.g. For "/home/runner/myFile.png", the fileBase is "myFile.png"
+
+In your terminal, you should see a ngrok URL, visiting this URL should display the LICENSE contents.
+
+### Part 3.3: Making requests to the Hackclub CDN API with Ky
+
+- [Ky](https://github.com/sindresorhus/ky) is an HTTP client that lets you easily make requests to any server.
+
+- We'll be using a POST request, which sends data to a server -- In this case, we'll be sending the ngrok URL.
+
+- Now, the Hackclub CDN API only accepts an array of URLs, so we'll need to wrap our URL in an array
+
+Add the following to `mod.ts`:
+
+```js
+import { connect, disconnect } from 'https://deno.land/x/ngrok@2.2.3/mod.ts'
+import { Application, send } from 'https://deno.land/x/oak@v6.3.1/mod.ts'
+import ky from 'https://unpkg.com/ky/index.js'
+import * as path from "https://deno.land/std@0.75.0/path/mod.ts"
+
+export default async function (filePath: string): Promise<void> {
+    const fileBase = path.parse(filePath).base
+
+    const app = new Application()
+    app.use(async (context: any) => {
+        await send(context, filePath, { root: '/' })
+    })
+
+    app.addEventListener("listen", async ({ port }) => {
+        const ngrokUrl = `https://${await connect({ protocol: 'http', port })}/${fileBase}`
+
+        // @ts-ignore
+        let response: string[] = await ky.post('https://cdn.hackclub.com/api/new', { json: [ngrokUrl] }).json() // <--- Wrapped in array [ ngrokUrl ]
+
+        console.log(response[0])     
+        disconnect()
+    })
+
+    await app.listen({ port: 20685 })
+}
+```
+
+Run `hcload` and you should get a URL like `https://cloud-something.vercel.app/yourFile.ext`. If opening the URL loads the file in your browser, or your browser asks you to download the file, CONGRATULATIONS! You've just hosted a file successfully on the CDN!
+
+Now, instead of logging the response from our library, we want to return it to the calling function. We'll log it from the calling function instead.
+
+```js
```

Comment:
I like the philosophy of "include context for what you're doing", but I think this is taking it a step too far. It's not clear to me at first glance what changed between this block of code and the previous block of code. I think instead of showing the whole file, which has a lot of things unchanged, just show a specific part of the file, or the function where the changed code is. Otherwise people will just copy/paste without thinking because they can't find the changes

Reply:
Quite a lot of the parts have been changed in the included code, and to clarify what changed I've already included a commend that says "`// <--- Changed`" on every line that has changed. Does this still need to be fixed?

Reply:
I think if a lot of parts of the code have changed, you should break each part down into its own section and then show the final changed version. The `//changed` comment is good but I think it's important to understand which parts changed and why at first glance